src/olm/session/mod.rs

```diff
@@ -599,6 +604,53 @@ mod test {
         Ok(())
     }
 
+    #[test]
+    fn max_keys_out_of_order_decryption() -> Result<()> {
```

Comment:
I know that we're using a `Result` for a bunch of tests in this repo, but this turns out to produce worse error messages if a test fails, compared to just panicking inside of the test.

Let's convert the `?` calls to `expect()`, or if they are trivial, to `unwrap()` calls.

Reply:
I have applied this change in two separate commits, one for the new tests I added and one for the existing tests in the same file. Some of the tests have gotten a little more verbose and I'm not sure where exactly to draw the line between using `expect` and `unwrap`. Happy to make further tweaks.